src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderCoinstakeRule.cs

```diff
@@ -241,6 +241,12 @@ private uint256 GetPreviousStakeModifier(ChainedHeader chainedHeader)
 
             if (previousProvenHeader != null)
             {
+                if (previousProvenHeader.StakeModifierV2 == null)
```

Comment:
I'm wondering if this check, or better the if at L242 should be done after the if below `if (chainedHeader.Previous.Height == 0)` L255

Reply:
probably even after `if (chainedHeader.Previous.Height == this.LastCheckpointHeight)` 
basically if we have a valid StakeModifierV2 (from genesis or checkpoint) we can even live without having a PH as a previous header, thoughts?

Reply:
Why? in the other cases stake modifier is provided by checkpoint or store